### PR TITLE
mighttpd2 3.2.4

### PR DESCRIPTION
**Pull request by pthariensflame**
Also updates and simplifies the formula.

### CODE REVIEW

File: Library/Formula/mighttpd2.rb [hunk 1]
```diff
@@ -16,14 +15,9 @@ class Mighttpd2 < Formula
 
   depends_on "ghc" => :build
   depends_on "cabal-install" => :build
-  depends_on "gmp"
```
Comment (mikemcquaid) [inline Library/Formula/mighttpd2.rb]: Are you sure? This runtime dependency seems to be for anything that depends on `ghc`.

Comment (mikemcquaid) [inline Library/Formula/mighttpd2.rb]: (trying to fix it in https://github.com/Homebrew/homebrew/pull/37479)

Comment (pthariensflame) [inline Library/Formula/mighttpd2.rb]: Oh!  I thought you had already merged that issue for some reason.  😕  I'll put the dependency back.

**Comment by pthariensflame** [comment]
@mikemcquaid How does this look?

**Comment by mikemcquaid** [comment]
All good, thanks @pthariensflame!

**Pull request closed without merging**